src/quantem/core/utils/augment_dp.py

```
@@ -551,6 +694,30 @@ def _apply_bkg(self, inputs: ArrayLike, probe: ArrayLike | None = None) -> Array
         inputs_float = af.as_type(inputs, torch.float32 if self.use_torch else np.float32)
         return inputs_float * (1 - self.bkg_weight) + CBEDbgConv.real * self.bkg_weight
 
+    def _apply_bkg_to_multichannel_label(self, label: ArrayLike, probe: ArrayLike | None = None) -> ArrayLike:
+        """Apply background to specified channels of multichannel label"""
+        if len(label.shape) != 3:
+            # Single channel label - shouldn't normally happen
```

should raise an error then

---

Added warning.